In [25]:
import pandas as pd
# from app2_refactored import DataManager

dataset_path = '/Users/artinmajdi/Documents/GitHubs/postdoc/Wound_management_interpreter_LLM/dataset/SmartBandage-Data_for_llm.csv'

df = pd.read_csv(dataset_path)
df.head()


Record ID                   Event Name         MRN Date of Birth     Sex  \
0          1     Visit 1 (Arm 2: Control)  41758670.0      6/9/1996  Female   
1          2     Visit 1 (Arm 2: Control)  41271032.0     1/23/1992    Male   
2          3  Visit 1  (Arm 1: Treatment)  41657125.0     8/29/1956    Male   
3          3   Visit 2 (Arm 1: Treatment)         NaN           NaN     NaN   
4          3   Visit 3 (Arm 1: Treatment)         NaN           NaN     NaN   

              Race  Race: Other - Specify               Ethnicity   Weight  \
0  White/ Caucasian                    NaN      Hispanic or Latino    93.3   
1  White/ Caucasian                    NaN      Hispanic or Latino    99.0   
2             Black                    NaN  Not Hispanic or Latino   111.2   
3               NaN                    NaN                     NaN     NaN   
4               NaN                    NaN                     NaN     NaN   

   Height   ...  Clinical events   \
0   157.48  ...               NaN   
1   182.00  ...               NaN   
2   180.34  ...               NaN   
3      NaN  ...               NaN   
4      NaN  ...               NaN   

  Diabetic Foot Wound - WIfI Classification: foot Infection (fI) Infection  \
0                                                NaN                   NaN   
1                                                NaN                   NaN   
2                                         Uninfected                   NaN   
3                                                NaN                   NaN   
4                                         Uninfected                   NaN   

   Infection/ Biomarker Measurement                  Granulation  \
0                                NaN                         NaN   
1                                NaN                         NaN   
2                                NaN  One half of the wound area   
3                                NaN                         NaN   
4                                NaN                         NaN   

  Granulation Quality         Necrosis  Exudate Volume Exudate Viscosity  \
0                  NaN              NaN            NaN               NaN   
1                  NaN              NaN            NaN               NaN   
2                 Pale  Adherent Slough           High            Medium   
3                  NaN              NaN            NaN               NaN   
4                  NaN              NaN         Medium            Medium   

  Exudate Type   
0           NaN  
1           NaN  
2        Yellow  
3           NaN  
4        Yellow  

[5 rows x 70 columns]

In [ ]:
df['Wound Type '].value_counts()

In [26]:
from llm_interface import WoundAnalysisLLM
from data_processor import WoundDataProcessor
llm = WoundAnalysisLLM(platform="ai-verde", model_name="llama-3.3-70b-fp8")
data_processor = WoundDataProcessor(df=df, dataset_path=dataset_path)

mode = 'all'
if mode=='all':
    patient_data = data_processor.get_population_statistics()
    prompt = llm._format_population_prompt(patient_data)
    # analysis = llm.analyze_population_data(patient_data)
else:
    patient_id = 3
    patient_data = data_processor.get_patient_visits(record_id=int(patient_id))
    prompt = llm._format_per_patient_prompt(patient_data)
    # analysis = llm.analyze_population_data(patient_data)

prompt


/Users/artinmajdi/Documents/GitHubs/postdoc/Wound_management_interpreter_LLM/wound_analysis/data_processor.py:115: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  'by_healing_status': df.groupby('BMI Category')['Healing Rate (%)'].agg(['mean', 'count']).to_dict()
/Users/artinmajdi/Documents/GitHubs/postdoc/Wound_management_interpreter_LLM/wound_analysis/data_processor.py:131: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  'diabetes_bmi': df.groupby(['Diabetes?', 'BMI Category'])['Healing Rate (%)'].agg(['mean', 'count']).to_dict()
/Users/artinmajdi/Documents/GitHubs/postdoc/Wound_management_interpreter_LLM/wound

"## Clinical Population Analysis Task\n\n**Context**: Analyze this wound care dataset from a smart bandage clinical trial. Focus on identifying clinically significant patterns, risk factors, and treatment outcomes.\n\n**Dataset Overview**:\n- Patients: N/A | Wounds: N/A\n- Study Duration: N/A | Avg Visits/Patient: N/A\n\n## 1. Patient Characteristics\n### Core Demographics:\n- Age: {'summary': 'Mean: 52.8, Median: 52.5', 'distribution': {52.0: 4, 42.0: 4, 67.0: 3, 62.0: 3, 30.0: 3, 65.0: 3, 44.0: 2, 71.0: 2, 70.0: 2, 43.0: 2, 64.0: 2, 74.0: 2, 56.0: 2, 76.0: 2, 37.0: 2, 38.0: 2, 59.0: 2, 51.0: 2, 26.0: 2, 75.0: 1, 36.0: 1, 63.0: 1, 72.0: 1, 80.0: 1, 53.0: 1, 47.0: 1, 50.0: 1, 28.0: 1, 68.0: 1, 58.0: 1, 82.0: 1, 24.0: 1, 34.0: 1, 57.0: 1, 48.0: 1, 31.0: 1, 33.0: 1, 78.0: 1, 2.0: 1}, 'age_groups': {'50-70': 28, '30-50': 19, '>70': 11, '<30': 8}} | Gender: {'Male': 38, 'Female': 28}\n- BMI: {'summary': 'Mean: 34.0, Range: 18.6-218.5', 'distribution': {'Obese': 30, 'Overweight': 18, 'Norma

In [27]:
patient_data.keys()

dict_keys(['summary', 'demographics', 'risk_factors', 'wound_characteristics', 'healing_progression', 'exudate_analysis', 'sensor_data'])

"## Clinical Population Analysis Task\n\n**Context**: Analyze this wound care dataset from a smart bandage clinical trial. Focus on identifying clinically significant patterns, risk factors, and treatment outcomes.\n\n**Dataset Overview**:\n- Patients: N/A | Wounds: N/A\n- Study Duration: N/A | Avg Visits/Patient: N/A\n\n## 1. Patient Characteristics\n### Core Demographics:\n- Age: {'summary': 'Mean: 52.8, Median: 52.5', 'distribution': {52.0: 4, 42.0: 4, 67.0: 3, 62.0: 3, 30.0: 3, 65.0: 3, 44.0: 2, 71.0: 2, 70.0: 2, 43.0: 2, 64.0: 2, 74.0: 2, 56.0: 2, 76.0: 2, 37.0: 2, 38.0: 2, 59.0: 2, 51.0: 2, 26.0: 2, 75.0: 1, 36.0: 1, 63.0: 1, 72.0: 1, 80.0: 1, 53.0: 1, 47.0: 1, 50.0: 1, 28.0: 1, 68.0: 1, 58.0: 1, 82.0: 1, 24.0: 1, 34.0: 1, 57.0: 1, 48.0: 1, 31.0: 1, 33.0: 1, 78.0: 1, 2.0: 1}, 'age_groups': {'50-70': 28, '30-50': 19, '>70': 11, '<30': 8}} | Gender: {'Male': 38, 'Female': 28}\n- BMI: {'summary': 'Mean: 34.0, Range: 18.6-218.5', 'distribution': {'Obese': 30, 'Overweight': 18, 'Normal': 15, 'Underweight': 0}, 'by_healing_status': {'mean': {'Underweight': nan, 'Normal': 0.0, 'Overweight': 0.0, 'Obese': 0.0}, 'count': {'Underweight': 0, 'Normal': 15, 'Overweight': 18, 'Obese': 30}}} | Race/Ethnicity: {'White/ Caucasian': 58, 'Black': 2, 'Other {race_oth}': 2, 'Two or more races': 2, 'Asian': 1, 'American Indian/ Alaskan Native': 1}\n\n### Comorbidity Profile:\n- Diabetes: N/A\n- Smoking: N/A\n- Vascular Disease: N/A\n\n## 2. Wound Characteristics\n### Typology:\n- Primary Types: N/A\n- Locations: N/A\n### Severity Metrics:\n- Initial Size: N/A\n- Depth Grades: N/A\n- Infection Rates: N/A\n\n## 3. Therapeutic Outcomes\n### Intervention Analysis:\n- Modalities Used: N/A\n- Response Times: N/A\n### Efficacy Metrics:\n- Healing Rates: N/A\n- Complication Profile: N/A\n\n## 4. Biomarker Correlations\n### Smart Bandage Metrics:\n- Temperature Trends: N/A\n- Impedance Dynamics: N/A\n- Oxygenation Levels: N/A\n\n### Healing Correlations:\n- Early Healing Indicators: N/A\n- Complication Predictors: N/A\n\n## Analysis Requirements\n1. Perform stratified analysis by wound type and patient comorbidities\n2. Identify significant correlations between sensor data and healing trajectories\n3. Evaluate treatment effectiveness with adjustment for confounding factors\n4. Provide clinical actionability scores for key findings\n5. Generate SMART clinical recommendations with implementation steps\n\n**Output Format**:\n- Structured clinical report with executive summary\n- Evidence-graded recommendations (Levels A-D)\n- Risk-benefit analysis for proposed interventions"


In [30]:
patient_data['sensor_data']

dict_keys(['oxygenation'])